File: app/src/main/java/cc/smartcash/wallet/Models/WebWalletRootResponse.java
package cc.smartcash.wallet.Models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

@JsonIgnoreProperties(ignoreUnknown = true)
public class WebWalletRootResponse<T> implements Serializable {

    @SerializedName("data")
    private T data;

    @SerializedName("error")
    private String error;

    @SerializedName("status")
    private String status;

    @SerializedName("isValid")
    private Boolean isValid;

    @SerializedName("error_description")
    private String errorDescription;

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    public String getError() {
        return error;
    }

    public void setError(String error) {
        this.error = error;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Boolean getValid() {
        return isValid;
    }

    public void setValid(Boolean valid) {
        isValid = valid;
    }

    public String getErrorDescription() {
        return errorDescription;
    }

    public void setErrorDescription(String errorDescription) {
        this.errorDescription = errorDescription;
    }

}

File: app/src/main/java/cc/smartcash/smarthub/Models/WebWalletRootResponse.kt
package cc.smartcash.smarthub.Models

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.google.gson.annotations.SerializedName

import java.io.Serializable

@JsonIgnoreProperties(ignoreUnknown = true)
data class WebWalletRootResponse<T>(
        @SerializedName("data")
        var data: T? = null,

        @SerializedName("error")
        var error: String? = null,

        @SerializedName("status")
        var status: String? = null,

        @SerializedName("isValid")
        var valid: Boolean? = null,

        @SerializedName("error_description")
        var errorDescription: String? = null
) : Serializable 
